### PR TITLE
ext_authz: code-quality improvements

### PR DESCRIPTION
**Pull request by LukeShu**
Commit Message:

Make various code-quality improvements around ext_authz:
- Fix a comment in the tests; the comment was written to an older revision of #11158 and had some untruths in
it.  And even if those things were true in the final version, it was confusingly worded.
- The `proto_format.sh` and friends have a lot of special cases for `envoy.service.auth.v2alpha` / `use_alpha`. Modify things to be less special. Now the only special thing about it should be a "magic" comment in `v3/external_auth.proto`.
- While I was at it figuring out how the proto_format stuff works, I noticed that `api_binding.bzl` says "envoy_api_raw" in a place where it surely means "envoy_api_canonical" (that is the only place in the whole repo where "envoy_api_raw" appears).  It doesn't seem to have been affecting anything.

Additional description: In order to facilitate reviewing human-made changes separately from machine-made changes, I ran `proto_format.sh` in a separate commit from the changes that triggered `proto_format.sh` to change things.
Risk Level: Low
Testing: With these commits applied on top of `v1.15.0` (instead of `master`): I've manually verified that `proto_format.sh` and friends still behave sanely, that `bazel test //test/...` still works, and that it passes Ambassador's test suite (verifying that `use_alpha` still behaves properly, since in the past there was a linker issue that caused it to not work in a way that the envoy.git tests didn't catch).
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

**Comment by dio** [comment]
cc. @lizan @htuch on the API stuff.

File: api/envoy/service/auth/v3/external_auth.proto [hunk 1]
```diff
@@ -27,6 +27,24 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
...
+  // with a comment:
+  //
+  //   previous_message_type = "envoy.service.auth.v2alpha.Authorization";
+  //
+  // Now, we'd also "normally" do that in the v2 package, not here.
+  // But, because the v2alpha package imports the v2 package, that'd
+  // cause a cycle.
+  //
+  // The lesson is that the "have v2alpha be an alias for v2 instead
+  // of duplicating the files" thing we did was a bad thing to do.
```
Comment (dio) [inline api/envoy/service/auth/v3/external_auth.proto:46]: Can we be more concise on this? Since this will be rendered in the docs, I think we need to consider how this can help an end-user.

File: test/extensions/filters/http/ext_authz/ext_authz_integration_test.cc [hunk 2]
```diff
@@ -182,10 +182,15 @@ class ExtAuthzGrpcIntegrationTest : public Grpc::VersionedGrpcClientIntegrationP
     }
 
     if (!new_headers_from_upstream.empty()) {
-      // new_headers_from_upstream has append = true. The current implementation ignores to set
-      // multiple headers that are not present in the original request headers. In order to add
-      // headers with the same key multiple times, setting response headers with append = false and
-      // append = true is required.
+      // new_headers_from_upstream has the `append=true` flag set, which tells Envoy to append to
+      // the existing header if it exists, and to discard the insertion otherwise. There's a
+      // "TODO(dio)" in ext_authz.cc to add a flag to make it be "append-if-set or set-if-unset",
```
Comment (dio) [inline test/extensions/filters/http/ext_authz/ext_authz_integration_test.cc:187]: Probably rephrase this to: In _method_name_ ...

And expand `ext_authz.cc` to full relative-path from root: source/extensions/filters/http/ext_authz/ext_authz.cc?

And I'm not sure a "TODO(dio)" is needed here. A pending task or a TODO is enough?

File: api/envoy/service/auth/v3/external_auth.proto (hunk 1 above)
Comment (LukeShu) [reply api/envoy/service/auth/v3/external_auth.proto:46]: Because it's not attached to a member of the service (there's a blank line after it), I don't think that's true.  `grep -r 'that the v2alpha package is' generated/docs/` gives me no results when I generate the docs locally.

File: test/extensions/filters/http/ext_authz/ext_authz_integration_test.cc (hunk 2 above)
Comment (LukeShu) [reply test/extensions/filters/http/ext_authz/ext_authz_integration_test.cc:187]: Whoops, I didn't think about that that reference would tag you.  My bad.

How's this?
```
// the existing header if it exists, and to discard the insertion otherwise. In
// `Filter::onComplete()` in `ext_authz.cc`, there's a "todo" to add a flag to make it be
// "append-if-set or set-if-unset", but for now it's just "set-if-unset".
```

File: api/envoy/service/auth/v2alpha/external_auth.proto [hunk 3]
```diff
@@ -2,12 +2,15 @@ syntax = "proto3";
 
 package envoy.service.auth.v2alpha;
 
+import "envoy/service/auth/v2/external_auth.proto";
```
Comment (htuch) [inline api/envoy/service/auth/v2alpha/external_auth.proto:5]: This a strange dependency relationship, having the v2alpha depend on the v2..

File: api/envoy/service/auth/v3/external_auth.proto [hunk 4]
```diff
@@ -27,6 +27,24 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
...
+  // For v2alpha to work as expected, we need to convince
+  // proto_sync.py that the v2alpha package is the previous version of
+  // either this package or the v2 package. And to do that, you'd
+  // "normally" use the
+  // "(udpa.annotations.versioning).previous_message_type" option; but
+  // we don't have a 'message' to put it on, we only have a 'service'.
+  // Fortunately, proto_sync.py's parser is sloppy, and we can fool it
+  // with a comment:
+  //
+  //   previous_message_type = "envoy.service.auth.v2alpha.Authorization";
```
Comment (htuch) [inline api/envoy/service/auth/v3/external_auth.proto:39]: While this trick may work, I think we're really relying on undocumented behavior in proto_sync.py. I'd prefer to keep the existing special cases, since they will sunset when v2 support is removed, there's not much benefit from optimizing for v2 consistency.

File: api/envoy/service/auth/v2alpha/external_auth.proto (hunk 3 above)
Comment (LukeShu) [reply api/envoy/service/auth/v2alpha/external_auth.proto:5]: Yeah, that strange relationship is the root reason for all the special cases.
